lta/rate_limiter.py

```diff
@@ -84,6 +84,18 @@ def _get_files_and_size(self, path: str) -> Tuple[List[str], int]:
         self.logger.info(f"Found {len(disk_files)} entries ({size} bytes) in {path}")
         return (disk_files, size)
 
+    def _get_files_and_size_with_retry(self, path: str) -> Tuple[List[str], int]:
+        """Recursively walk and add the files of files in the file system."""
+        retry_left = 3
```

Comment:
I can think of two causes for a failure:  
1) The file isn't there any more--which is a good thing, and we should just skip on to the next bundle
2) The filesystem is having issues--a bad thing, which will eventually manifest as a failure to move the bundle from staging to output
The former doesn't need to raise an exception, and the latter will do so later.  I'd think a simple try/except around the filesystem accesses on lines 64 and 82 would be better than raising an exception when the file is missing.

Reply:
Something like:
        for disk_file in disk_files:
            # determine the size of the file
            try:
                size += os.path.getsize(disk_file)
            except:
                # File gone already?  OK.
                continue
            self.logger.debug(f"Size so far: {size} bytes")